src/spetlr/deltaspec/DeltaTableSpecDifference.py

```diff
@@ -0,0 +1,356 @@
+                + " NOT NULL"
+            )
+
+        for key, newComment in alter_comment:
+            statements.append(
+                f"ALTER TABLE {self.base.name} ALTER COLUMN "
+                f"{key} COMMENT {json.dumps(newComment)}"
+            )
+
+        # Finally address the possible change of order.
+
+        intermediate_target_order = copy.copy(base_order)
+        for field in fields_to_add:
+            intermediate_target_order.append(field.name)
+
+        # now the intermediate_target_order always contains
+        # the same fields in the same order as the table for now.
+        first = target_order[0]
+        if intermediate_target_order[0] != first:
+            intermediate_target_order.pop(intermediate_target_order.index(first))
+            intermediate_target_order.insert(0, first)
+            statements.append(
+                f"ALTER TABLE {self.base.name} ALTER COLUMN {first} FIRST"
+            )
+
+        for i, key in enumerate(target_order[1:], 1):
+            if intermediate_target_order[i] != key:
+                intermediate_target_order.pop(intermediate_target_order.index(key))
+                intermediate_target_order.insert(i, key)
+                statements.append(
+                    f"ALTER TABLE {self.base.name} ALTER COLUMN {key} AFTER "
+                    + target_order[i - 1]
+                )
+
+        # possible differences:
+        # - different capitalization -- Not handled. Treated as new column
+        # - different nullability -- Done
+        # - different struct members (new, dropped)
+        #           -- Not handled. Treated as new column
+        # - different comment
+        # - different order
+
+        # Todo: FIX THEM ALL
+
+        return statements
+
+    def _alter_statements_for_new_location(self, allow_new_columns=False) -> List[str]:
+        """In case we need to change location,
+        the alter statements will consist of creating the new target.
+        and then pointing the current table at it. The target will be empty."""
+
+        if not self.target.location:
+            raise NotImplementedError(
+                "Converting an external to a managed table not implemented"
+            )
+
+        # Note: This will fail if the target is not empty
+
+        # Create the target table, then point the current table at it
+        nameless_target = self.target.fully_substituted(name=None)
+        statements = [nameless_target.get_sql_create()]
+        statements.append(
+            f"ALTER TABLE {self.base.name} SET LOCATION {json.dumps(self.target.name)}"
+        )
+        # now to old base name points at the new location.
+        # all that remains is to perhaps rename it
+        new_base = nameless_target.fully_substituted(name=self.base.name)
+
+        statements += self.target.compare_to(new_base).alter_table_statements(
+            allow_new_columns=allow_new_columns
+        )
+        return statements
```

Comment:
I dont get how this works...
1: Create a nameless target... alright.
2: create the sql statemetn... alright.
3: alter the current table to point to the new tables location... that does not seem smart right?
4: Then the nameless target is tried to be called the same as the base - but the base is not dropped right? so here it will encounter an issue...

what then happens with the old table? 

Is this tested?

Reply:
The old table data is not touched. It is left in the old location. Cleanup is up to the user.